go-controller/pkg/ovn/pods.go

```diff
@@ -262,43 +237,45 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 		// If the pod already has annotations use the existing static
 		// IP/MAC from the annotation.
+		addresses = fmt.Sprintf("%s %s", podMac, podCIDR.IP)
 		args = append(args,
-			"--", "lsp-set-addresses", portName, fmt.Sprintf("%s %s", podMac, podCIDR.IP),
+			"--", "lsp-set-addresses", portName, addresses,
 			"--", "--if-exists", "clear", "logical_switch_port", portName, "dynamic_addresses",
 		)
 	} else {
-		addresses := "dynamic"
+		podMac, podCIDR, err = oc.assignPodAddresses(logicalSwitch)
+		if err != nil {
+			return fmt.Errorf("Failed to assign pod address: %s", err)
+		}
+		addrAssigned = true
+
 		networks, err := util.GetPodNetSelAnnotation(pod, util.DefNetworkAnnotation)
 		if err != nil || (networks != nil && len(networks) != 1) {
-			return fmt.Errorf("error while getting custom MAC config for port %q from "+
-				"default-network's network-attachment: %v", portName, err)
+			return handleError(logicalSwitch, &podCIDR.IP, addrAssigned, fmt.Errorf("error while getting custom MAC config for port %q from "+
+				"default-network's network-attachment: %v", portName, err))
 		} else if networks != nil && networks[0].MacRequest != "" {
 			klog.V(5).Infof("Pod %s/%s requested custom MAC: %s", pod.Namespace, pod.Name, networks[0].MacRequest)
-			addresses = networks[0].MacRequest + " dynamic"
+			podMac, err = net.ParseMAC(networks[0].MacRequest)
+			if err != nil {
+				return handleError(logicalSwitch, &podCIDR.IP, addrAssigned, fmt.Errorf("Failed to parse mac %s requested in annotation for pod %s: Error %s", networks[0].MacRequest, pod.Name, err))
+			}
 		}
 
+		addresses = fmt.Sprintf("%s %s", podMac, podCIDR.IP)
+
 		// If it has no annotations, let OVN assign it IP and MAC addresses
```

Comment:
Should this code still exist?

Reply:
I am guessing we still need to set the addresses on the lsp? If there is no consumer in the nbdb/northd, we can get rid of it. @dcbw?